### ISSUE TITLE
Incredible waiting time when typing TypeScript code in Visual Studio

### ISSUE THREAD

**Issue by Doomisateur** [microsoft/TypeScript#29024]
Hi,

I got a problem since last year. I waited for new version on VS and typescript but it didn't solved it.

My intellisense is perfect then 2 min after coding it's very slow. 10-20 seconds before proposition appear.

I saw a lot of thread about this problem. 

I haven't any weard extension and I tried to disable the basic one by one without difference.

When my problem apears NodeJs is consuming 600Mo~1Go of memory and I have 4 instances on my task manager.

The problem happens immedialty, it's not progressive. It's perfect then very slow. The problem is not occurs on C# part.

Windows: 10
Visual studio: Pro 2017
Current TypeScript version 3.1 but I tested with others before (from 2.1 to 3.1)
NodeJS version : 10.6.0

Thanks for your help
![sans titre](https://user-images.githubusercontent.com/17436539/50011112-84655200-ffbb-11e8-90d5-5d3ce39e966a.png)
![sans titre2](https://user-images.githubusercontent.com/17436539/50011113-84655200-ffbb-11e8-841a-190e4b24b875.png)

**Comment by MgSam** [comment]
Probably not what you want to hear, but I'd recommend using VSCode for the TypeScript portions of your application. VSCode is fast, stable, and gets new updates every month. Once we can write all our .NET code in VSCode we'll probably move all development over to it.

Visual Studio is a 2nd class editor for TypeScript and it looks like that situation will only get worse over time as I think usage numbers of TypeScript in Visual Studio are extremely low. Neglect begets neglect.

**Comment by sheetalkamat** [comment]
@Doomisateur 
[enabled logging in VS](https://billti.github.io/jsdocs/articles/troubleshooting/index.html#enable-detailed-language-service-logging) and provide us the log to be able to investigate this.

**Comment by Doomisateur** [comment]
Thank you for your answers. I prefer a solution compared to VS2017, I do not intend to use a new product on which I will have to meet other problems. Especially since I can not believe that I am the only one to meet this problem, there is inevitably a solution.
Here is in attachment the log activated on a day.
Thank you and good day
[tsserver.zip](https://github.com/Microsoft/TypeScript/files/2683477/tsserver.zip)

**Comment by sheetalkamat** [comment]
@Doomisateur From the tsserver log there wasn't a single completion request that took more than 600ms. Is the log from right session? I also notice you are using 2.8.4 tsserver. Can you try using tsserver@next and see if things are better? Thanks

**Comment by Doomisateur** [comment]
@sheetalkamat Thank you for your answer.
When I generated this log, I had a lot of slowdown however in Visual Studio 2017.
However, you make me notice that I use TypeScript 3.1 and yet you notice in the logs that it is a version 2.8 which is used. Maybe my problem would be at that level?
I just checked, when I go in the properties of my project, I select the version 3.1, it is still the 2.8 version that seems to be active when I return.
After that, I checked my NuGet packages and I could see that my TypeScript package is still at version 2.8.
As a result, I will update it to the latest version. I will generate new logs and I will keep you informed if my latency problem persists.
Greetings
![typescript_v](https://user-images.githubusercontent.com/17436539/50162783-6a8b7e00-02df-11e9-914f-20f77d3a2232.png)
![typescript_nuget](https://user-images.githubusercontent.com/17436539/50162913-ae7e8300-02df-11e9-9f9a-ef2eccee417d.png)

**Comment by sheetalkamat** [comment]
Adding @minestarks and @amcasey who would know more about selecting typescript version in VS

**Comment by Doomisateur** [comment]
Since this change, I have not yet seen a slowdown. All indications are that the problem was the "bad version of the NuGet package". Thank you very much for your help, you allowed me to see that I was using a bad version of TypeScript. With a good configuration on version 3.1, everything seems to be in order.
However I want to keep this ticket open until the end of the week because unfortunately I had very little time to test everything.
Thank you for your answers.

**Comment by amcasey** [comment]
@Doomisateur I didn't understand this comment:

> I just checked, when I go in the properties of my project, I select the version 3.1, it is still the 2.8 version that seems to be active when I return.

Do you mean that you set the option but it still showed the old value when you returned to the options page later?

**Comment by amcasey** [comment]
It sounds like switching to TS 3.1 (possibly in a way that mitigated a VS bug around version selection) resolved the issue.  Please let us know if you continue to see pauses during editing.

**Comment by Doomisateur** [comment]
Hello,
When I encountered these slowness problems, I was on an earlier version of Windows. However I had a failure on my computer and on my Windows at the beginning of the year which forced me to install a new version of Windows 10 and therefore also of my entire development environment. And since then, I have never encountered any problems with slowness on the new installation.
Sorry, I hadn't seen your January 2 message. Yes it was exactly that, even when I went to version 3.1, it went back to version 2.8 when I came back in the page.
Have a good day
